packages/coreutils/src/poll.ts

```diff
@@ -7,6 +7,22 @@ import { IDisposable } from '@phosphor/disposable';
 
 import { ISignal, Signal } from '@phosphor/signaling';
 
+/**
+ * A function to defer an action immediately.
+ */
+const defer =
+  typeof requestAnimationFrame === 'function'
+    ? requestAnimationFrame
+    : setImmediate;
+
+/**
+ * A function to cancel a deferred action.
+ */
+const cancel: (timeout: any) => void =
+  typeof cancelAnimationFrame === 'function'
+    ? cancelAnimationFrame
+    : clearImmediate;
+
```

Comment:
This is cribbed from PhosphorJS: https://github.com/phosphorjs/phosphor/blob/master/packages/messaging/src/index.ts#L490-